src/Nethermind/Nethermind.Facade/BlockchainBridge.cs

```diff
@@ -187,10 +207,19 @@ public SimulateOutput Simulate(BlockHeader header, SimulatePayload<TransactionWi
                 result.Error = ex.ToString();
             }
 
-            result.Items = simulateOutputTracer.Results;
+            if (tracer is SimulateBlockTracer simulateTracer)
+            {
+                result.Items = simulateTracer.Results as IReadOnlyList<SimulateBlockResult<T>>;
+            }
```

Comment:
Instead of this make generic constraint `where T : ISimulateTracer` and `IBlockTracer<TTrace>` interface should have `Results` of this type.